### DISCUSSION TITLE
How can I use usingComponent in modal?

### DISCUSSION THREAD

**Discussion by mirbaagheri** [orchidsoftware/platform#2856] [Q&A]
Hi,
I'm trying to use `usingComponent` in a modal.
But I have this error: `Orchid\Screen\Components\Cells\Currency::__construct(): Argument #1 ($value) must be of type float, null given`
My code:
```Layout::modal('new-order-modal', [
                Layout::legend('', [
                    Sight::make('product.price','Price:')->usingComponent(Currency::class),
                    Sight::make('email'),
                ]),
                
                ])
                ->async('asyncGetProduct')
                ->applyButton('Buy Now'),```

**Comment by mirbaagheri** [comment]
@tabuna

**Comment by tabuna** [comment]
The error clearly indicates that you are attempting to display `null` as a monetary value, which conflicts with the expectations of the `Currency` component that requires a `float` type.

I would recommend setting the default value for the corresponding database column to `0` and disallowing `null`. However, if you have intentionally set it up this way in the database, you can add an [accessor](https://laravel.com/docs/11.x/eloquent-mutators) in your model to convert `null` to an explicit typed value, such as `0.00`. This will help avoid errors and ensure proper data display.

**Reply by mirbaagheri** [reply]
You misunderstood the problem or my explanation was not enough.
I am going to use the `Currency` class in defining a modal.
The modal opens when I click on its button. Therefore, in first time, there is no value for `price` , and this error occurs exactly for this reason.

```
Layout::modal('new-order-modal', [
                Layout::legend('', [
                    Sight::make('product.price','Product Price:')->usingComponent(Currency::class),
                    ])->border(false)->class('no-py new-order-modal'),
                ])
                ->async('asyncGetProduct')
                ->applyButton('Buy Now'),
```
Now, when I open the page, I get the following error:
`Orchid\Screen\Components\Cells\Currency::__construct(): Argument #1 ($value) must be of type float, null given`

If I make third line as comment and then open my page, I dont jave any error. after it If I uncomment third line and come back to the desired page(without refresh the page) the modal will work fine.

https://github.com/user-attachments/assets/9bdd4428-414d-40c2-9813-a2fbd1285d0e

**Comment by mirbaagheri** [comment]
Poor support.
